### ISSUE TITLE
Swapping registry context/pools

### ISSUE THREAD

**Issue by Casqade** [skypjack/entt#980]
I was thinking about implementing a background loading routine and found out there's no builtin way to swap `basic_registry` pools while preserving context variables. E.g. deserialize entities state into a temporary registry in separate thread & move its contents into my main registry when it's ready. 

At this moment, my idea can be implemented with something like this:
```
void Game::onLoadComplete( entt::registry& loadedRegistry )
{
  this->mainRegistry.swap(loadedRegistry);
  std::swap( this->mainRegistry.ctx(), loadedRegistry.ctx() );
}
```
However, I'm not sure whether it's a valid or convenient way to do it...

I can create a PR with `basic_registry::swap_ctx` and `basic_registry::swap_pools` which basically just execute different parts of `basic_registry::swap` (`swap_pools` is a misleading name tbh: it basically swaps everything except context vars. Maybe it's redundant anyway) . 
Although I'm not sure if I'll manage to come up with a proper test for these methods.

**Comment by skypjack** [comment]
Well, yeah, when you swap two elements, the whole content is passed around. Context variables are part of it. 🤷‍♂️ 
If you know the types that you've loaded, you can still swap the pools of interest as in:
```cpp
mainRegistry.storage<T>().swap(loadedRegistry.storage<T>());
```
As for adding these very specific functions to the registry API, I'm usually against it.
I don't like updating an API with functions that aim to solve a specific user problem when these can be implemented externally.
Your `onLoadComplete` looks just fine in fact and it doesn't affect the API of the registry, so win-win, right? 🙂

**Comment by Casqade** [comment]
> when these can be implemented externally

Good point, I'm also quite conservative when it comes to introducing such routines into any API. However, I encounter an opposite point of view from time to time. 
Anyway, I'm closing the issue then, thank you!
